cms/src/test/java/uk/nhs/digital/common/OrderedSearchDateDerivedDataFunctionTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.nhs.digital.common;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import org.apache.jackrabbit.value.BooleanValue;
+import org.apache.jackrabbit.value.DateValue;
+import org.apache.jackrabbit.value.LongValue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.jcr.Value;
+
+@RunWith(DataProviderRunner.class)
+public class OrderedSearchDateDerivedDataFunctionTest {
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy");
+    private static final String ORDERED_SEARCH_DATE = "orderedSearchDate";
+
+    private OrderedSearchDateDerivedDataFunction function = new OrderedSearchDateDerivedDataFunction();
+
+    @Test
+    public void testUpcoming() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("20-03-2020", false);
+
+        long expectedValue = -parseDate("20-03-2020").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
```

Comment:
This and other test methods could more immediately communicate what is the set up here and what is the actual action that is being tested.

How about changing the order of operations to follow the given/when/then sequence, thus making it more explicit that `testFunctionCompute` leads to the the actual action that we're testing here?

Adding comments `// given`, `// when`, `// then` would further clarify purpose of each block, especially that the `testFunctionCompute` method makes it less obvious that the actual call under test is the `function.compute(...)` one.